File: POBJ/Csharp/Ex_4_Reprise_Cpp/Ex_4_Reprise_Cpp/Cercle.cs
using System;

public class Cercle : FormesGeometriques
{
    // Constructeur
    public Cercle()
    {
        Console.WriteLine("Constructeur de Cercle");
    }

    public override double CalculerSurface(double diametre, double unused)
    {
        double rayon = diametre * 0.5;  //diamètre /2
        Surface = Math.PI * rayon * rayon;
        return Surface;
    }

    public override double CalculerPerimetre(double diametre, double unused)
    {
        Perimetre = Math.PI * diametre;
        return Perimetre;
    }
}

File: POBJ/CPP/Exercices/EX_4/EX_4/Cercle.cpp
#define _USE_MATH_DEFINES

#include "Cercle.h"
#include <cmath>

//Constructeur
Cercle::Cercle()
    {
    std::cout << "Constructeur de Cercle" << std::endl;
    }

//Destructeur
Cercle::~Cercle()
{
    std::cout << "Destructeur de Cercle" << std::endl;
}


double Cercle::CalculerSurface(double diametre, double null)
{
    double rayon = diametre * 0.5;  //diamètre /2
    Surface = M_PI * rayon * rayon;
    return Surface;
}

double Cercle::CalculerPerimetre(double diametre, double null)
{
    Perimetre = M_PI * diametre;
    return Perimetre;
}
